guides/keras_nlp/quick_tour.py

```diff
@@ -0,0 +1,610 @@
+    weighted_metrics=keras.metrics.SparseCategoricalAccuracy(),
+    jit_compile=True,
+)
+
+# Pretrain on IMDB dataset
+pretraining_model.fit(
+    pretrain_ds,
+    validation_data=pretrain_val_ds,
+    epochs=3,  # Increase to 6 for higher accuracy
+)
+
+"""
+After pretraining save your `backbone` submodel to use in a new task!
+"""
+
+"""
+# Build and train your own transformer from scratch
+![picture](https://drive.google.com/uc?id=1pzwLPCtvzmHY3DKzH-MBzmjWFJ3pKVB5)
+
+Want to implement a novel transformer architecture? The `keras-nlp` library offers all
+the low-level modules used to build SoTA architectures in our `models` API. This includes
+training your own subword tokenizer using `WordPiece`, `BytePairEncoder`, or
+`SentencePiece`.
+
+In this workflow we train a custom tokenizer on the IMDB data and design a backbone with
+custom transformer architecture. For simplicity we then train directly on the
+classification task. Interested in more details? We wrote an entire guide to pretraining
+and finetuning a custom transformer:
+https://keras.io/guides/keras_nlp/transformer_pretraining/
+"""
+
+"""
+## Train custom vocabulary from IMBD data
+"""
+
+vocab = keras_nlp.tokenizers.compute_word_piece_vocabulary(
+    imdb_train.map(lambda x, y: x),
+    vocabulary_size=10_000,  # Increase to 20_000 for better performance
+    lowercase=True,
+    strip_accents=True,
+    reserved_tokens=["[PAD]", "[START]", "[END]", "[MASK]", "[UNK]"],
+)
+tokenizer = keras_nlp.tokenizers.WordPieceTokenizer(
+    vocabulary=vocab,
+    lowercase=True,
+    strip_accents=True,
+    oov_token="[UNK]",
+)
+
+"""
+
+## Preprocess data with custom tokenizer
+"""
+
+packer = keras_nlp.layers.StartEndPacker(
+    start_value=tokenizer.token_to_id("[START]"),
+    end_value=tokenizer.token_to_id("[END]"),
+    pad_value=tokenizer.token_to_id("[PAD]"),
+    sequence_length=64,
+)
+
+
+def preprocess(x, y):
+    token_ids = packer(tokenizer(x))
+    x = {
+        "token_ids": token_ids,
+        "padding_mask": token_ids != tokenizer.token_to_id("[PAD]"),
+    }
+    return x, y
+
+
+imdb_preproc_train_ds = imdb_train.map(
+    preprocess, num_parallel_calls=tf.data.AUTOTUNE
+).prefetch(tf.data.AUTOTUNE)
+imdb_preproc_val_ds = imdb_test.map(
+    preprocess, num_parallel_calls=tf.data.AUTOTUNE
+).prefetch(tf.data.AUTOTUNE)
+
+imdb_preproc_train_ds.unbatch().take(1).get_single_element()
+
+"""
+
+## Design a tiny transformer
+"""
+
+token_id_input = keras.Input(
+    shape=(None,),
+    dtype="int32",
+    name="token_ids",
+)
+padding_mask = keras.Input(
+    shape=(None,),
+    dtype="int32",
+    name="padding_mask",
+)
+outputs = keras_nlp.layers.TokenAndPositionEmbedding(
+    vocabulary_size=len(vocab),
+    sequence_length=packer.sequence_length,
+    embedding_dim=64,
+)(token_id_input)
+outputs = keras_nlp.layers.TransformerEncoder(
+    num_heads=2,
+    intermediate_dim=128,
+    dropout=0.1,
+)(outputs, padding_mask=padding_mask)
```

Comment:
`TokenAndPositionEmbedding` and `TransformerEncoder` aren't mask aware? I'd expect to be able to do

```python
token_id_input = keras.Input(
    shape=(None,),
    dtype="int32",
    name="token_ids",
)
outputs = keras_nlp.layers.TokenAndPositionEmbedding(
    vocabulary_size=len(vocab),
    sequence_length=packer.sequence_length,
    embedding_dim=64,
    mask_zero=True,
)(token_id_input)
outputs = keras_nlp.layers.TransformerEncoder(
    num_heads=2,
    intermediate_dim=128,
    dropout=0.1,
)(outputs)
```

(assuming input sequences have zeros at padded locations)

Reply:
They are mast aware. We could do this and it would simplify the code a bit.